packages/connect-iframe/src/index.ts

```diff
@@ -24,17 +24,30 @@ import {
 import { Core, init as initCore, initTransport } from '@trezor/connect/src/core';
 import { DataManager } from '@trezor/connect/src/data/DataManager';
 import { config } from '@trezor/connect/src/data/config';
-import { initLog } from '@trezor/connect/src/utils/debug';
+import { LogWriter, createLogger } from '@trezor/connect/src/utils/debug';
 import { getOrigin } from '@trezor/connect/src/utils/urlUtils';
 import { suggestBridgeInstaller } from '@trezor/connect/src/data/transportInfo';
 import { suggestUdevInstaller } from '@trezor/connect/src/data/udevInfo';
 import { storage, getSystemInfo, getInstallerPackage } from '@trezor/connect-common';
 import { parseConnectSettings, isOriginWhitelisted } from './connectSettings';
+import LogWorker from './sharedLoggerWorker';
 
 let _core: Core | undefined;
 
+const logWorker = new LogWorker();
+logWorker.port.start();
+
+const logWriter = (): LogWriter => {
+    return {
+        add: (message: object) => logWorker.port.postMessage({ type: 'add-log', data: message }),
+    };
+};
+
+const initLog = createLogger(logWriter);
+
```

Comment:
@mroz22 This approach works, here and in https://github.com/trezor/trezor-suite/pull/8623/files#diff-39f16e786cb0eac481161e3a249c4ade088cc38cf0bdd8adc74ac7b7bac99172R31

I think this could work also in connect core.

What do you think?

Reply:
here on line 261 there is 
```
_core = await initCore(parsedSettings);
```
I think that we don't lose anything if we extend initCore with another parameter (writer). Iframe will pass its own, default writer here, and other environments where serialization does not happen will be able to pass their own writer directly from src/index